lib/rucio/tests/test_bin_rucio.py

```diff
@@ -1525,3 +1526,116 @@ def test_list_account_usage(self):
         assert re.search('.*{0}.*{1}.*{2}.*{3}'.format(rse_exp, usage, global_limit, global_left), out) is not None
         self.account_client.set_local_account_limit(account, rse, -1)
         self.account_client.set_global_account_limit(account, rse_exp, -1)
+
+    def test_upload_recursive_ok(self):
```

Comment:
Nice! 
would be a good bonus if you could adapt the `file_factory` fixture added by Mayank and used in test_upload.py to generate the folders and files as needed for your new tests; then use it here. We are gradually migrating to using pytest and fixtures to avoid code duplication in tests.

Reply:
I have adapted the `file_factory` fixture but I am unable to use it inside `TestBinRucio` because it inherits `unittest.TestCase` and according to the [docs](https://pytest.org/en/latest/how-to/unittest.html): 

> unittest.TestCase methods cannot directly receive fixture arguments as implementing that is likely to inflict on the ability to run general unittest.TestCase test suites.

Reply:
Hi. Thank you for doing this work. It will allow to easier migrate these tests to pytest when we'll do it. In the meantime you can just use the class directly like this https://github.com/rucio/rucio/blob/5d1085b03e376307fa33b94b52a25fc8f77fc9e0/lib/rucio/tests/conftest.py#L167 .